### PR TITLE
task_in_progress_changes

### PR DESCRIPTION
**Pull request by prabhakaranE6157**
Pull Request type
----
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes (Please run `./gradlew generateLock saveLock` to refresh dependencies)
- [ ] WHOSUSING.md
- [ ] Other (please describe):

**NOTE**: Please remember to run `./gradlew spotlessApply` to fix any format violations.

Changes in this PR
----

_Describe the new behavior from this PR, and why it's needed_
Issue #

Alternatives considered
----

_Describe alternative implementation you have considered_

### CODE REVIEW

File: scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java [hunk 1]
```diff
@@ -87,7 +90,7 @@ public class ScyllaExecutionDAO extends ScyllaBaseDAO
 
     protected final PreparedStatement selectWorkflowsByCorIdFromWorkflowStatement;
 
-    protected final PreparedStatement selectCountFromTaskInProgressStatement;
+    //protected final PreparedStatement selectCountFromTaskInProgressStatement;
```
Comment (geethalladi) [inline scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java]: Can you please delete these commented blocks ? If required we can revert this one

**Comment by geethalladi** [comment]
@prabhakaranE6157 Should we update the tests for this ?

File: scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java [hunk 2]
```diff
@@ -333,38 +347,71 @@ public List<TaskModel> createTasks(List<TaskModel> tasks) {
...
+        return session.execute(selectTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId())));
+    }
+
+    private ResultSet fetchFromTaskInProgressQuery(TaskModel task) {
+        logTaskInProgressStatus(task);
+        return session.execute(selectTaskInProgressStatement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId())));
+    }
 
+    private boolean isResultSetNotValid(ResultSet resultSet) {
+        return Objects.isNull(resultSet) || resultSet.all().size() < 1;
```
Comment (kartik2k) [inline scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java]: resultSet.all().isEmpty()

File: scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java [hunk 3]
```diff
@@ -333,38 +347,71 @@ public List<TaskModel> createTasks(List<TaskModel> tasks) {
...
     }
 
     /**
      * @method to remove the task_in_progress table with the status of the task
      */
     public void removeTaskInProgress(TaskModel task) {
-        session.execute(
-                deleteTaskInProgressStatement.bind(task.getTaskDefName(),UUID.fromString(task.getTaskId())));
+        ResultSet resultSet = session.execute(
+                deleteTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId())));
```
Comment (kartik2k) [inline scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java]: Have we verified what happens if the row itself is not present and we attempt to delete it? Specifically, does the result.wasApplied set return true or false in such a scenario?
If Yes then we might need to run 2 queries.

File: scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java [hunk 4]
```diff
@@ -333,38 +347,71 @@ public List<TaskModel> createTasks(List<TaskModel> tasks) {
...
 
     /**
      * @method to update the task_in_progress table with the status of the task
      */
     public void updateTaskInProgress(TaskModel task, boolean inProgress) {
-        session.execute(
-                updateTaskInProgressStatement.bind(inProgress,task.getTaskDefName(),UUID.fromString(task.getTaskId())));
+        ResultSet resultSet =
+                session.execute(updateTaskInProgressV2Statement.bind(inProgress, task.getTaskDefName(), UUID.fromString(task.getTaskId())));
+        if (!resultSet.wasApplied()) {
```
Comment (kartik2k) [inline scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java]: same please check here too

**Review by kartik2k — CHANGES_REQUESTED** [review]

File: scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java (hunk 3 above)
Comment (prabhakaranE6157) [reply scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java]: `resultSet.wasApplied()` is wrong here, putting both in batch, similar for other

**Pull request merged**